internal/tracker/source/tekton/tekton.go

```diff
@@ -348,23 +352,50 @@ func PreparePackage(i *PreparePackageInput) (*hub.Package, error) {
 	var name, version, description, tektonKind string
 	var annotations map[string]string
 	var tasks []map[string]interface{}
+	var containerImages []*hub.ContainerImage
+
 	switch m := i.Manifest.(type) {
 	case *v1beta1.Task:
 		tektonKind = "task"
 		name = m.Name
 		version = m.Labels[versionLabelTKey]
 		description = m.Spec.Description
 		annotations = m.Annotations
+
+		// Prepare container images
+		ts := m.TaskSpec()
+		var defaults []v1beta1.ParamSpec
+		if len(ts.Params) > 0 {
+			defaults = append(defaults, ts.Params...)
+		}
+		mts := task.ApplyParameters(context.Background(), &ts, &v1beta1.TaskRun{}, defaults...)
+		for _, s := range mts.Steps {
+			if _, err := n.ParseReference(s.Image); s.Image != "" && err == nil {
+				c := hub.ContainerImage{Image: s.Image}
+				containerImages = append(containerImages, &c)
+			}
+		}
 	case *v1beta1.Pipeline:
 		tektonKind = "pipeline"
 		name = m.Name
 		version = m.Labels[versionLabelTKey]
 		description = m.Spec.Description
 		annotations = m.Annotations
-		for _, task := range m.Spec.Tasks {
+
+		// Prepare container images
+		ps := m.PipelineSpec()
+		mps := p.ApplyParameters(context.Background(), &ps, &v1beta1.PipelineRun{})
+		for _, mts := range mps.Tasks {
+			for _, ms := range mts.TaskSpec.Steps {
```

Comment:
Is it possible that steps in different tasks in a pipeline use the same image? I don't think this is checked at a higher level at the moment (and it should, I'll take a note to review/improve it), but it'd be great if we can make sure they are not duplicated here for now. Otherwise it's likely that all copies will be processed and displayed in the security report.

Reply:
Yeah, this is a good point. I checked other kinds and didn't see a dedup logic there so I thought it is fine. But sure will add a dedup logic here :)

Reply:
Awesome, thanks. For the other kinds it hasn't been a problem so far. But I suspect this is more likely to happen in Tekton pipelines.

Reply:
Added dedup logic!